### PR TITLE
bug fixes and logic improvements for windows scripts

### PR DESCRIPTION
**Pull request by rosskirkpat**
#### Proposed Changes ####

- bug fixes for the logic in the rke2 rancher install script that attempts retries when the v3 endpoint is not yet populated
- improved logic for cleaning up the machine and user/process variables in the windows uninstall script
- added semi-advanced error handling to the `get-rke2config` function in the rke2 rancherinstall script
- the containerd process is now stopped after attempting to remove all of it's resources to avoid a potential race condition
- ctr calls during cleanup use a nested `Invoke-Expression` function to join args properly
- containerd cleanup now has default namespaces to use if it doesn't detect any 
- processes are now stopped in a function at the appropriate time during uninstall script
- the logic/process of cleaning up processes and services has been updated based on feedback from @phillipsj 

#### Types of Changes ####

bug fix and code improvments

#### Verification ####

I manually ran the install/uninstall scripts to verify that they worked and the bugs were fixed.

Screenshot of rancher v2.6.1-rc13 with two windows nodes added using the updated install script

1st Screenshot of running the updated uninstall script

2nd Screenshot of running the updated uninstall script

There are 2 screenshots as the network connection was reset when the HNS networks were modified. This step is being moved to the end to prevent needing to connect to the node a second time to complete the clean process. This is an expected behavior.

#### Linked Issues ####

https://github.com/rancher/rke2/issues/1948
https://github.com/rancher/rke2/issues/1935

#### Further Comments ####

One outstanding item that is left in a TODO state is to use crictl to cleanup pods in addition to the existing ctr cleanup
https://github.com/rancher/rke2/issues/1950

### CODE REVIEW

File: bundle/bin/rke2-uninstall.ps1 [hunk 1]
```diff
@@ -101,29 +101,20 @@ function Invoke-HNSRequest {
...
-    Write-LogInfo "Stopping process $($_.Name) ..."
-    $_ | Stop-Process -ErrorAction Ignore -Force
-}
-
-Get-Process -ErrorAction Ignore -Name "wins*" | ForEach-Object {
-    Write-LogInfo "Stopping process $($_.Name) ..."
-    $_ | Stop-Process -ErrorAction Ignore -Force
+function Stop-Processes () {
+
+    $ProcessNames = @('rke2', 'kube-proxy', 'kubelet', 'containerd', 'wins')
```
Comment (luthermonson) [inline bundle/bin/rke2-uninstall.ps1:106]: dont we want a calico too?

File: bundle/bin/rke2-uninstall.ps1 [hunk 2]
```diff
@@ -253,34 +220,65 @@ function Remove-TempData () {
...
     catch {
-        Write-LogWarn "Could not reset machine environment variables: $($_)"
+        Write-LogWarn "Could not reset machine environment variable: $($_)"
     }
 }
+
 function Remove-Containerd () {
-    if (Test-Command ctr) {
+    $CONTAINERD_ADDRESS = "\\.\\pipe\\containerd-containerd"
+    crictl config --set runtime-endpoint=npipe:\\.\\pipe\\containerd-containerd
```
Comment (luthermonson) [inline bundle/bin/rke2-uninstall.ps1:264]: why not use the var above on this line?

File: bundle/bin/rke2-uninstall.ps1 [hunk 3]
```diff
@@ -101,29 +101,20 @@ function Invoke-HNSRequest {
...
-Get-Process -ErrorAction Ignore -Name "containerd*" | ForEach-Object {
-    Write-LogInfo "Stopping process $($_.Name) ..."
-    $_ | Stop-Process -ErrorAction Ignore -Force
-}
-
-Get-Process -ErrorAction Ignore -Name "wins*" | ForEach-Object {
-    Write-LogInfo "Stopping process $($_.Name) ..."
-    $_ | Stop-Process -ErrorAction Ignore -Force
+function Stop-Processes () {
+
```
Comment (luthermonson) [inline bundle/bin/rke2-uninstall.ps1:105]: nit...

**Review by luthermonson — APPROVED** [review]

File: bundle/bin/rke2-uninstall.ps1 (hunk 3 above)
Comment (rosskirkpat) [reply bundle/bin/rke2-uninstall.ps1]: removed the new line

**Pull request merged**